### DISCUSSION TITLE
HSS build process vs `hss-payload-generator`

### DISCUSSION THREAD

**Discussion by pcrosemurgy** [polarfire-soc/polarfire-soc-documentation#504] [HSS and U-Boot] [answered]
Reading the top-level HSS [`README.md`](https://github.com/polarfire-soc/hart-software-services/tree/master?tab=readme-ov-file#building) I see how the HSS binary is built and flashed to eNVM.

However, I'm having trouble understanding how the [`hss-payload-generator`](https://github.com/polarfire-soc/hart-software-services/tree/master/tools/hss-payload-generator) relates to this top-level build process. From `hss-payload-generator/README.md` it sounds like it produces a binary that can also be flashed to eNVM. Is there a high-level explanation for how these two build processes relate to one another?

It seems like the `hss-payload-generator` must be used if you want to flash an HSS with a custom `u-boot.bin` payload to eNVM?

Thanks!

**Comment by hughbreslin** [comment, marked as answer]
Hey @pcrosemurgy

The HSS runs from eNVM, due to the size of the eNVM not much else can fit in there with it (128KB). The HSS payload generator creates payloads with headers which can be booted from external non-volatile memory, like eMMC, SD cards etc. The payload generator is not part of the HSS build process but is actually part of the embedded software build process.

I would suggest having a look at this [document](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/knowledge-base/polarfire-soc-software-tool-flow.md#polarfire-soc-software-tool-flow) on the tool flow and also this [document](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/hss-and-u-boot/hss-payloads.md) which describes HSS payloads.
